hare/algorithm.go

```diff
@@ -117,15 +113,18 @@ func (proc *ConsensusProcess) Start() error {
 	return nil
 }
 
-func (proc *ConsensusProcess) createInbox(size uint32) chan *pb.HareMessage {
-	proc.inbox = make(chan *pb.HareMessage, size)
-	return proc.inbox
+func (proc *ConsensusProcess) Id() InstanceId {
+	return proc.instanceId
+}
+
+func (proc *ConsensusProcess) SetInbox(inbox chan *pb.HareMessage) {
+	proc.inbox = inbox
```

Comment:
just to be safe, consider validating that `proc.inbox` is nil. Otherwise, print an error

Reply:
Also, in the Start method, you should validate the the inbox isn't nil

Reply:
Fixed